pydantic/fields.py

```diff
@@ -39,14 +40,132 @@
     from .class_validators import ValidatorCallable  # noqa: F401
     from .error_wrappers import ErrorList
     from .main import BaseConfig, BaseModel  # noqa: F401
-    from .schema import Schema  # noqa: F401
     from .types import ModelOrDc  # noqa: F401
 
     ValidatorsList = List[ValidatorCallable]
     ValidateReturn = Tuple[Optional[Any], Optional[ErrorList]]
     LocStr = Union[Tuple[Union[int, str], ...], str]
 
 
+class FieldInfo:
+    """
+    Captures extra information about a field.
+    """
+
+    __slots__ = (
+        'default',
+        'alias',
+        'title',
+        'description',
+        'const',
+        'gt',
+        'ge',
+        'lt',
+        'le',
+        'multiple_of',
+        'min_items',
+        'max_items',
+        'min_length',
+        'max_length',
+        'regex',
+        'extra',
+    )
+
+    def __init__(self, default: Any, **kwargs: Any) -> None:
+        self.default = default
+        self.alias = kwargs.pop('alias', None)
+        self.title = kwargs.pop('title', None)
+        self.description = kwargs.pop('description', None)
+        self.const = kwargs.pop('const', None)
+        self.gt = kwargs.pop('gt', None)
+        self.ge = kwargs.pop('ge', None)
+        self.lt = kwargs.pop('lt', None)
+        self.le = kwargs.pop('le', None)
+        self.multiple_of = kwargs.pop('multiple_of', None)
+        self.min_items = kwargs.pop('min_items', None)
+        self.max_items = kwargs.pop('max_items', None)
+        self.min_length = kwargs.pop('min_length', None)
+        self.max_length = kwargs.pop('max_length', None)
+        self.regex = kwargs.pop('regex', None)
+        self.extra = kwargs
+
+    def __repr__(self) -> str:
+        attrs = ((s, getattr(self, s)) for s in self.__slots__)
+        return 'FieldInfo({})'.format(', '.join(f'{a}: {v!r}' for a, v in attrs if v is not None))
+
+
+def Field(
```

Comment:
Just want to put this on the radar before we commit to this approach:
https://github.com/samuelcolvin/pydantic/issues/837